src/librbd/image/OpenRequest.cc

```diff
@@ -277,7 +277,48 @@ Context *OpenRequest<I>::handle_v2_get_immutable_metadata(int *result) {
                << cpp_strerror(*result) << dendl;
     send_close_image(*result);
   } else {
-    send_v2_get_stripe_unit_count();
+      send_v2_get_features();
+  }
+
+  return nullptr;
+}
+
+template <typename I>
+void OpenRequest<I>::send_v2_get_features() {
+  CephContext *cct = m_image_ctx->cct;
+  ldout(cct, 10) << this << " " << __func__ << dendl;
+
+  librados::ObjectReadOperation op;
+  cls_client::get_features_start(&op);
+
+  using klass = OpenRequest<I>;
+  librados::AioCompletion *comp = create_rados_callback<
+    klass, &klass::handle_v2_get_features>(this);
+  m_out_bl.clear();
+  m_image_ctx->md_ctx.aio_operate(m_image_ctx->header_oid, comp, &op,
+                                  &m_out_bl);
+  comp->release();
+}
+
+template <typename I>
+Context *OpenRequest<I>::handle_v2_get_features(int *result) {
+  CephContext *cct = m_image_ctx->cct;
+  ldout(cct, 10) << __func__ << ": r=" << *result << dendl;
+
+  if (*result == 0) {
+    bufferlist::iterator it = m_out_bl.begin();
+    *result = cls_client::get_features_finish(
+      &it, &m_image_ctx->features);
+  }
+  if (*result < 0) {
+    lderr(cct) << "failed to retreive features: "
+               << cpp_strerror(*result) << dendl;
+    send_close_image(*result);
+  } else {
```

Comment:
Nit: just return ```nullptr``` if you need to close the image and then eliminate the nested ```if``` block